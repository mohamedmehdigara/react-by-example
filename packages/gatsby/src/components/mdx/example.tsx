import React, { FC, ReactNode, useCallback, useState } from 'react'
import styled from 'styled-components'
import { CodeSandboxIcon, ResetIcon } from '../icons'

const Render = styled.div`
  display: flex;
  padding: 1em;
  border: 1px solid var(--app-foreground);
  border-radius: var(--app-border-radius);
  align-items: center;
  justify-content: center;
`
const FlexRight = styled.div`
  display: flex;
  justify-content: flex-end;
`

const CodeSandboxLink = styled.a`
  display: flex;
`

const ResetButton = styled.button`
  display: flex;
  margin: 0;
  padding: 0;
  color: var(--app-foreground);
  background: none;
  border: none;
  cursor: pointer;

  :focus,
  :active {
    color: var(--app-cyan);
    outline: 2px solid var(--app-cyan);
  }
`

export interface ExampleProps {
  code?: ReactNode
  render?: ReactNode
  link?: string
}

export const Example: FC<ExampleProps> = ({ code, render, link }) => {
  const [resetKey, setResetKey] = useState(Date.now())
  const triggerReset = useCallback(() => {
    setResetKey(Date.now())
  }, [])
  return (
    <>
      {code}
      <Render key={resetKey}>
        <div>{render}</div>
      </Render>
      {link && (
        <FlexRight>
          <ResetButton
            type="button"
            title="reset example"
            onClick={triggerReset}
          >
            <ResetIcon />
          </ResetButton>
          <CodeSandboxLink
            target="_blank"
            rel="noreferrer"
            href={link}
            title="edit on CodeSandbox"
          >
            <CodeSandboxIcon />
          </CodeSandboxLink>
        </FlexRight>
      )}
    </>
  )
}
