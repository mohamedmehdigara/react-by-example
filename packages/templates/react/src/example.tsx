import React, { FC, ReactElement } from 'react'

export const Simple: FC = () => <h1>Hello world!</h1>

export default (): ReactElement => <Simple />
