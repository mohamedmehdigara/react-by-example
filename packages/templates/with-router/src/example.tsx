import React, { FC } from 'react'

export const Simple: FC = () => <h1>Hello world!</h1>

export default <Simple />
