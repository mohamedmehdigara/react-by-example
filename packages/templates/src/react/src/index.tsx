import React from 'react'
import ReactDOM from 'react-dom'
import example from './example'

ReactDOM.render(<>{example}</>, document.getElementById('root'))
