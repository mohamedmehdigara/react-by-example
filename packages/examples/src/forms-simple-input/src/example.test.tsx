import { cleanup, render, RenderResult } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import React from 'react'
import { SimpleInput } from './example'

describe('SimpleInput', () => {
  let component: RenderResult
  let input: HTMLElement
  let button: HTMLElement
  let display: HTMLElement

  beforeEach(() => {
    component = render(<SimpleInput />)
    input = component.getByLabelText('simple input')
    button = component.getByRole('button')
    display = component.getByText('default value')
  })

  afterEach(() => {
    cleanup()
  })

  it('should snapshot', () => {
    expect(component.container).toMatchSnapshot()
  })

  it('should have default value', () => {
    expect(display).toHaveTextContent('default value')
    expect(input).toHaveValue('default value')
  })

  describe('when typing in input', () => {
    beforeEach(async () => {
      await userEvent.type(input, '{selectall}should update')
    })

    it('should update', () => {
      expect(display).toHaveTextContent('should update')
      expect(input).toHaveValue('should update')
    })

    describe('when setting value programmatically', () => {
      beforeEach(() => {
        userEvent.click(button)
      })

      it('should update', () => {
        expect(display).toHaveTextContent('new value')
        expect(input).toHaveValue('new value')
      })
    })
  })
})
