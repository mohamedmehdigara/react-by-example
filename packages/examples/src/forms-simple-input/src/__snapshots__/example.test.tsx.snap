// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleInput should snapshot 1`] = `
<div>
  <div>
    <label>
      simple input
      <input
        type="text"
        value="default value"
      />
    </label>
    <pre>
      default value
    </pre>
    <button
      type="button"
    >
      change to \`new value\`
    </button>
  </div>
</div>
`;
